clippy_lints/src/erasing_op.rs

```diff
@@ -0,0 +1,62 @@
+use consts::{constant_simple, Constant};
+use rustc::hir::*;
+use rustc::lint::*;
+use syntax::codemap::Span;
+use utils::{in_macro, span_lint};
+
+/// **What it does:** Checks for erasing operations, e.g. `x * 0`.
+///
+/// **Why is this bad?** The whole expression can be replaced by zero.
+/// Most likely mistake was made and code should be reviewed or simplified.
```

Comment:
I'd rather write "This is most likely not the intended outcome and should probably be corrected."